daemon/src/db/cull_dlcs.rs

```diff
@@ -0,0 +1,187 @@
+use crate::db::Connection;
+use anyhow::Result;
+use model::EventKind;
+
+impl Connection {
+    #[allow(dead_code)]
+    async fn cull_old_dlcs(&self) -> Result<()> {
+        let mut conn = self.inner.acquire().await?;
+
+        let rows = sqlx::query!(
+            r#"
+SELECT
+    id,
+    name,
+    data
+FROM
+    events
+WHERE
+    name = $1 OR
+    name = $2
+ORDER BY
+    created_at DESC
+LIMIT -1 OFFSET 1
+            "#,
+            model::EventKind::CONTRACT_SETUP_COMPLETED_EVENT,
+            model::EventKind::ROLLOVER_COMPLETED_EVENT,
+        )
+        .fetch_all(&mut conn)
+        .await?;
+
+        for row in rows.into_iter() {
+            let id = row.id;
+            let data = EventKind::from_json(row.name, row.data)?;
+
+            let event = match data {
+                EventKind::ContractSetupCompleted { dlc: Some(_), .. } => {
+                    EventKind::ContractSetupCompleted { dlc: None }
+                }
+                EventKind::RolloverCompleted {
+                    dlc: Some(_),
+                    funding_fee,
+                } => EventKind::RolloverCompleted {
+                    dlc: None,
```

Comment:
I'm not sure this is safe: we need the old `revocation` and `publish` keys to be able to punish.

Reply:
This should all stored here:

https://github.com/itchysats/itchysats/blob/0a7b03d39dd5c9d6e483bde31d4fae14163a4dd3/model/src/cfd.rs#L1811

https://github.com/itchysats/itchysats/blob/0a7b03d39dd5c9d6e483bde31d4fae14163a4dd3/model/src/cfd.rs#L2000-L2015

Reply:
Ah, thanks for looking this up.

Reply:
@luckysori does that mean we actually stored this information redundantly in the past? 
Comes in handy now 😅

Reply:
Yes, we had multiple copies of this, growing after each rollover.